### PR TITLE
REP-3123 Default HTTP connection pool should be used when unknown id is configured

### PR DESCRIPTION
**Pull request by dmnjohns**
I removed the deprecated container configuration in this story. I also changed the HTTP connection pool service api a bit. Oh, and I moved the configuration classes to the api module rather than the impl module.

### CODE REVIEW

**Comment by Mario-Lopez** [comment]
retest this please

**Comment by Mario-Lopez** [comment]
retest this please

File: repose-aggregator/components/services/http-client-service/http-client-service-impl/src/test/groovy/org/openrepose/core/services/httpclient/impl/HttpConnectionPoolServiceImplTest.groovy [hunk 1]
```diff
@@ -50,97 +52,109 @@ class HttpConnectionPoolServiceImplTest {
...
     }
 
-    @Test
-    void testGetClient() {
-        HttpClient client = srv.getClient(POOL1_ID).getHttpClient();
-        assertEquals("Should retrieve requested client", POOL1_SO_TIMEOUT, client.getParams().getParameter(CoreConnectionPNames.SO_TIMEOUT));
+    @After
+    void destroy() {
+        srv = new HttpConnectionPoolServiceImpl(configurationService, healthCheckService)
+        srv.destroy()
```
Comment (Mario-Lopez) [inline repose-aggregator/components/services/http-client-service/http-client-service-impl/src/test/groovy/org/openrepose/core/services/httpclient/impl/HttpConnectionPoolServiceImplTest.groovy]: I'm not sure why you would create a new instance to call destroy on it after every unit test.

File: repose-aggregator/components/services/http-client-service/http-client-service-impl/src/test/groovy/org/openrepose/core/services/httpclient/impl/HttpConnectionPoolServiceImplTest.groovy [hunk 2]
```diff
@@ -50,97 +52,109 @@ class HttpConnectionPoolServiceImplTest {
     // Retrieve the defaults defined in the XSD.
     PoolType poolType = new PoolType()
     Integer POOLU_SO_TIMEOUT = poolType.getHttpSocketTimeout()
```
Comment (Mario-Lopez) [inline repose-aggregator/components/services/http-client-service/http-client-service-impl/src/test/groovy/org/openrepose/core/services/httpclient/impl/HttpConnectionPoolServiceImplTest.groovy]: Unused variable.

**Comment by Mario-Lopez** [comment]
Missed a reference to the old config:  https://github.com/rackerlabs/repose/blob/REP-3123/repose-aggregator/functional-tests/spock-functional-test/src/test/configs/features/core/proxy/clientRequestLoggingTrue/container.cfg.xml#L3

File: repose-aggregator/commons/commons-utilities/src/main/java/org/openrepose/commons/utils/http/ServiceClient.java [hunk 3]
```diff
@@ -80,7 +79,7 @@ public ServiceClient(String targetHostUri, String username, String password, Str
     }
 
     private HttpClient getClientWithBasicAuth() throws ServiceClientException {
```
Comment (Mario-Lopez) [inline repose-aggregator/commons/commons-utilities/src/main/java/org/openrepose/commons/utils/http/ServiceClient.java]: Is throwing this exception still necessary?  If not, we may not need the exception class anymore.

**Comment by Mario-Lopez** [comment]
Other than the comments and 1 unexpected test failure, stuff looks good.

`features.filters.herp.AuthHerpDerpRMSTest.Req with auth resp: 401`

File: repose-aggregator/components/services/http-client-service/http-client-service-impl/src/test/groovy/org/openrepose/core/services/httpclient/impl/HttpConnectionPoolServiceImplTest.groovy (hunk 1 above)
Comment (dmnjohns) [inline repose-aggregator/components/services/http-client-service/http-client-service-impl/src/test/groovy/org/openrepose/core/services/httpclient/impl/HttpConnectionPoolServiceImplTest.groovy]: I... have no idea how that got there.

File: repose-aggregator/components/services/http-client-service/http-client-service-impl/src/test/groovy/org/openrepose/core/services/httpclient/impl/HttpConnectionPoolServiceImplTest.groovy (hunk 2 above)
Comment (dmnjohns) [inline repose-aggregator/components/services/http-client-service/http-client-service-impl/src/test/groovy/org/openrepose/core/services/httpclient/impl/HttpConnectionPoolServiceImplTest.groovy]: Yarp.

File: repose-aggregator/commons/commons-utilities/src/main/java/org/openrepose/commons/utils/http/ServiceClient.java (hunk 3 above)
Comment (dmnjohns) [inline repose-aggregator/commons/commons-utilities/src/main/java/org/openrepose/commons/utils/http/ServiceClient.java]: Good call. I don't see the exception being constructed anywhere, so I'll remove it and make sure everything still works.

File: repose-aggregator/commons/commons-utilities/src/main/java/org/openrepose/commons/utils/http/ServiceClientException.java [hunk 4]
```diff
@@ -1,31 +0,0 @@
...
- */
-package org.openrepose.commons.utils.http;
-
-import org.openrepose.core.services.httpclient.HttpClientNotFoundException;
-
-/**
- * Custom exception to handle apache http client exceptions in Serviceclient
- */
-public class ServiceClientException extends Throwable {
-    public ServiceClientException(String message, HttpClientNotFoundException e) {
```
Comment (Mario-Lopez) [inline repose-aggregator/commons/commons-utilities/src/main/java/org/openrepose/commons/utils/http/ServiceClientException.java]: Does the HttpClientNotFoundException class need to exist anymore?

**Pull request merged**